pkg/scanner/clair/scanner.go

```diff
@@ -55,23 +55,24 @@ func (s *imageScanner) Scan(req harbor.ScanRequest) (harbor.ScanResponse, error)
 	return harbor.ScanResponse{ID: layerName}, nil
 }
 
-func (s *imageScanner) prepareLayers(req harbor.ScanRequest) ([]clair.Layer, error) {
-	layers := make([]clair.Layer, 0)
-
-	registryClient, err := s.registryClientFactory.Get(req.Registry.URL, req.Registry.Authorization)
+func (s *scanner) prepareLayers(req harbor.ScanRequest) ([]clair.Layer, error) {
+	registryClient, err := s.registryClientFactory.Get(req)
```

Comment:
Do we need to create a registry client for each scan request?  Is it better to use a singleton client to handle the requests?

Reply:
One adapter may be used by different harbor instances, so registry client should be created from scan request.

Reply:
I think it's a design consideration. Base on the current case, a singleton client is ok.

Reply:
Good point. I refactored so that `registry.ClientFactory.Get()` returns singleton client preconfigured with TLS settings inferred from environment variables. At the same time keeping the code unit test friendly.